src/averageAges.js

```diff
@@ -55,7 +61,16 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = people.filter(child => (onlyWithSon
+    ? people.find(mother => child.mother === mother.name && child.sex === 'm')
+    : people.find(mother => child.mother === mother.name)));
+
+  const difference = children.map(child =>
+    (child.born - people.find(mother => mother.name === child.mother).born));
```

Comment:
separate `people.find(mother => mother.name === child.mother)` to function and pass `motherName` for better readability.

```suggestion
    (child.born - findMother(child.mother).born));
```